src/pal/pal_noalloc.h

```diff
@@ -4,23 +4,29 @@
 #pragma once
 
 #include "../aal/aal.h"
+#include "pal_concept.h"
 #include "pal_consts.h"
 
 #include <cstring>
 
 namespace snmalloc
 {
+#ifdef __cpp_concepts
+  /**
+   * The minimal subset of a PAL that we need for delegation
+   */
+  template<typename PAL>
+  concept PALNoAllocBase = ConceptPAL_static_sizes<PAL>&& ConceptPAL_error<PAL>;
+#endif
+
   /**
    * Platform abstraction layer that does not allow allocation.
    *
    * This is a minimal PAL for pre-reserved memory regions, where the
    * address-space manager is initialised with all of the memory that it will
    * ever use.
-   *
-   * It takes an error handler delegate as a template argument. This is
-   * expected to forward to the default PAL in most cases.
    */
-  template<typename ErrorHandler>
+  template<SNMALLOC_CONCEPT(PALNoAllocBase) BasePAL>
```

Comment:
Yes, but I think the `PALNoAllocBase` is a restricted subset and so is fine.  The important thing is that the base PAL-like thing shouldn't be required to be able to allocate memory because accidentally passing through `notify_not_using` on a shared memory segment may not do the right thing.